packages/flutter_test/test/widget_tester_test.dart

```diff
@@ -894,6 +898,42 @@ void main() {
         .checkMockMessageHandler(SystemChannels.accessibility.name, null), isTrue);
     });
   });
+
+  group('Leak tracker catches that', () {
+    // These tests cannot run inside other tests because test nesting is forbidden.
+    // So, `expect` happens outside the tests, in `tearDown`.
+    late Leaks leaks;
+
+    testWidgets(
+      '$StatelessLeakingWidget leaks',
+      (WidgetTester tester) async {
+        await tester.pumpWidget(StatelessLeakingWidget());
+      },
+      trackMemoryLeaks: true,
+      leakTrackingFlutterTestConfig: LeakTrackingFlutterTestConfig(
+        onLeaks: (Leaks theLeaks) {
+          leaks = theLeaks;
+        },
+        failTestOnLeaks: false,
+      ),
+    );
+
+    tearDown(() {
+      expect(() => expect(leaks, isLeakFree), throwsException);
```

Comment:
maybe check that it throws the right exception (e.g. that it has the expected message) to make sure we are not catching an unrelated exception here.